### PR TITLE
cmake: Exclude some targets

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref #6201

Problem Summary:

### What is changed and how it works?

1. Exclude most targets from All.

   Number of builds when building the All target (`ninja`):

   - Before this PR: 3928
   - This PR: 3366
   - Baseline (`ninja tiflash`): 3360

2. No need to execute post_install scripts for debug profiles. We only care about the binary size in release profile.

Extracted from https://github.com/pingcap/tiflash/pull/6112

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Server/CMakeLists.txt [hunk 1]
```diff
@@ -125,7 +125,9 @@ if (OS_LINUX)
                 DESTINATION ".")
     endif()
 
-    install (SCRIPT ${TiFlash_SOURCE_DIR}/cmake/tiflash_linux_post_install.cmake COMPONENT tiflash-release)
+    if (NOT (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG" OR SAN_DEBUG))
+        install (SCRIPT ${TiFlash_SOURCE_DIR}/cmake/tiflash_linux_post_install.cmake COMPONENT tiflash-release)
+    endif()
```
Comment (JaySon-Huang) [inline dbms/src/Server/CMakeLists.txt:130]: I think we can keep this under debug mode as the tiflash binary is too large under debug mode. And reducing the binary size can speedup our CI.

Comment (breezewish) [reply dbms/src/Server/CMakeLists.txt:130]: Makes sense for me.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**